Control/test/lib/middleware/mocha-addDetectorId.middleware.test.js

```diff
@@ -0,0 +1,57 @@
+const assert = require('assert');
+const sinon = require('sinon');
+const express = require('express');
+const request = require('supertest');
+
+// Import the middleware
+const { addDetectorIdMiddleware } = require('../../../lib/middleware/addDetectorId.middleware');
+
+describe('`addDetectorIdMiddleware` test suite', () => {
+  it('should add the specified detectorId to req.params and call next()', () => {
+    const detectorId = '1234';
+    const req = { params: {} }; // Mock req object
+    const res = {}; // Mock res object
+    const next = sinon.stub(); // Mock next function
+
+    addDetectorIdMiddleware(detectorId)(req, res, next);
+
+    // Validate the detectorId is added
+    assert.strictEqual(req.params.detectorId, detectorId);
+    // Ensure next() is called
+    assert.ok(next.calledOnce);
+  });
+
+  it('should overwrite existing detectorId in req.params', () => {
+    const detectorId = '5678';
+    const req = { params: { detectorId: 'original' } };
+    const res = {};
+    const next = sinon.stub();
+
+    addDetectorIdMiddleware(detectorId)(req, res, next);
+
+    // Validate the detectorId is overwritten
+    assert.strictEqual(req.params.detectorId, detectorId);
+    assert.ok(next.calledOnce);
+  });
+
+  it('should integrate with an Express route and add detectorId', async () => {
+    const app = express();
```

Comment:
This test is not in-line with the standards of the GUI testing. In this file, we should have only tests for the middleware function and not integrated in a server. 
Moreover, the server is just a usual one and not even the WebUI one. For API integrated services with have a separate `test/api` folder where we test the entire routes. Such tests should go in there and not here.
Could you please remove this test for now and when it comes to API testing we will add it there in a different PR

Reply:
oké will remove it,